src/main/java/com/singlestore/jdbc/plugin/codec/ByteArrayCodec.java

```diff
@@ -49,6 +50,11 @@ public boolean canEncode(Object value) {
     return value instanceof byte[];
   }
 
+  @Override
+  public int getApproximateTextProtocolLength(Object value) throws SQLException {
+    return canEncode(value) ? ((byte[]) value).length * 2 : -1;
```

Comment:
Why do we multiply it here?

Reply:
looking to encoding method writeBytesEscaped - it required 2*length buffer when adding escape slash